Sources/Basics/MultiFormatArchiver.swift

```diff
@@ -0,0 +1,111 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import struct TSCBasic.AbsolutePath
+
+/// An `Archiver` that handles multiple formats by delegating to other archivers it was initialized with.
+public final class MultiFormatArchiver: Archiver {
+    public var supportedExtensions: Set<String>
+
+    enum Error: Swift.Error {
+        case unknownFormat(String, AbsolutePath)
+        case noFileNameExtension(AbsolutePath)
+
+        var description: String {
+            switch self {
+            case .unknownFormat(let ext, let path):
```

Comment:
Why not consider using magic? https://github.com/file/file/tree/ec41083645689a787cdd00cb3b5bf578aa79e46c should be possible to use on Windows even.

Reply:
Can we assume it's installed on Windows? It doesn't seem to be installed on `swift` Docker images for Ubuntu, so adding a dependency on it will require updating `Dockerfile`s in the `swift-docker` repository. This isn't supposed to be an incredibly resilient algorithm, but something that we can still land in 5.9 I hope. So distinguishing by extension should be good enough for now, and we can consider adding support for magic format detection later.